### PR TITLE
CLI command for repacking extensions

### PR DESCRIPTION
**Pull request by kasimi**
There are two issues I haven't been able to figure out so far:

* When passing a message with `-m "Message"` the Extensions Robot creates a post in the validation discussion topic. However, its post count isn't increased.

* I tried to fix generating routes as explained [here](http://symfony.com/doc/current/console/request_context.html) but it doesn't seem to work. Specifically, the `["extra"]["version-check"]` object of the `revision_composer_json` value in the database contains wrong data after running the command:
  * `"host"` is `null`
  * `"directory"` has the absolute path on the server prepended: `/32/db/extension/post_numbers` becomes `/vagrant/www/32/db/extension/post_numbers`

Any hints on how to get those issues fixed?

### CODE REVIEW

File: language/en/console.php [hunk 1]
```diff
@@ -44,4 +44,14 @@
 	'CLI_DESCRIPTION_REBUILD_COMPOSER'		=> 'Rebuild Composer repository',
 	'CLI_REBUILD_COMPOSER_FORCE'			=> 'Force rebuild even if a build is in progress',
 	'CLI_REBUILD_COMPOSER_FROM_FILE'		=> 'Rebuild repository from revision zip files',
+
+    'CLI_DESCRIPTION_EXTENSION_REPACK'      => 'Repack extension revisions',
+    'CLI_EXTENSION_REPACK_EXT_NAME'         => 'vendor/extname to repack. If omitted, all revisions of all extensions are repacked.',
+    'CLI_EXTENSION_REPACK_MESSAGE'          => 'Message to post in each extension’s validation discussion topic.',
```
Comment (Elsensee) [inline language/en/console.php]: Here you've used spaces instead of tabs.

File: console/command/extension/repack.php [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+	 *
+	 * @param titania_config	$titania_config
+	 * @param type				$ext_type
+	 * @param attachments		$attachments
+	 */
+	public function set_titania_dependencies(
+		titania_config $titania_config,
+		type $ext_type,
+		attachments $attachments
+	)
```
Comment (Elsensee) [inline console/command/extension/repack.php]: I'd personally prefer it to have all the arguments in a single line, actually..

**Comment by kasimi** [comment]
Regarding the first issue (Extensions Robot's post count not increasing): the post is created with type `ext::TITANIA_QUEUE_DISCUSSION` but in the config there's only [`ext::TITANIA_SUPPORT`](https://github.com/phpbb/customisation-db/blob/6599973510cf89c42137479faf0ff2d1864fb680/config/config.php#L60) enabled to increase the post count. I don't think it's problem really.

I'm still investigating the wrong host/directory...

**Comment by DavidIQ** [comment]
Yeah not worried about the robot account's post count not going up. This path thing looks familiar though...I'll think about it for a bit.

**Comment by DavidIQ** [comment]
Not coming up with anything. @paul999 do you remember the path issue from before?

**Comment by paul999** [comment]
Nope, can't remember seeing it

**Comment by VSEphpbb** [comment]
kasimi are you testing in a 3.2.x board or 3.3.x/master? You should be testing within 3.2.x.

**Comment by kasimi** [comment]
I'm testing on a 3.2.0 board.

**Comment by kasimi** [comment]
Got this fixed finally. [The route helper creates a new request context on each call](https://github.com/phpbb/phpbb/blob/release-3.2.1/phpBB/phpbb/routing/helper.php#L104-L114) which is why modifying the `router` service had no effect. Setting `force_server_vars` to true was the solution.

Will the arguments have to be moved into a single line? I think it's more readable when they are on separate lines.

File: console/command/extension/repack.php [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+		$this->template			= $template;
+		$this->db				= $db;
+
+		$this->contribs_table	= TITANIA_CONTRIBS_TABLE;
+		$this->revisions_table	= TITANIA_REVISIONS_TABLE;
+
+		$this->language->add_lang(array('console', 'manage', 'manage_tools', 'contributions'), 'phpbb/titania');
+
+		// As we're in the CLI, we need to force server vars so that the route helper generates correct version check URLs
+		$this->config['force_server_vars'] = true;
```
Comment (Nicofuma) [inline console/command/extension/repack.php]: I think this will alter the board configuration in database.

**Comment by Nicofuma** [comment]
I have one concern with the $config modification, otherwise looks good to me. And sorry for the delay

**Comment by kasimi** [comment]
[offsetSet()](https://github.com/phpbb/phpbb/blob/release-3.2.1/phpBB/phpbb/config/config.php#L69-L81) changes the value for the current request only. For a persistent change, [set()](https://github.com/phpbb/phpbb/blob/release-3.2.1/phpBB/phpbb/config/db.php#L112-L123) would have to be used.

No worries about the delay!

**Comment by DavidIQ** [comment]
Is that last comment regarding @Nicofuma's board configuration comment in `repack.php`?

**Comment by kasimi** [comment]
Ah, yes.

**Pull request merged**